pkg/interceptors/bitbucket/bitbucket_test.go

```diff
@@ -0,0 +1,264 @@
+		},
+		{
+			name: "valid header for secret, but no matching event",
+			Bitbucket: &triggersv1.BitBucketInterceptor{
+				SecretRef: &triggersv1.SecretRef{
+					SecretName: "mysecret",
+					SecretKey:  "token",
+				},
+				EventTypes: []string{"pr:opened", "repo:refs_changed"},
+			},
+			args: args{
+				// This was generated by using SHA1 and hmac from go stdlib on secret and payload.
+				// https://play.golang.org/p/otp1o_cJTd7 for a sample.
+				signature: "sha1=38e005ef7dd3faee13204505532011257023654e",
+				secret: &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "mysecret",
+					},
+					Data: map[string][]byte{
+						"token": []byte("secret"),
+					},
+				},
+				eventType: "event",
+				payload:   ioutil.NopCloser(bytes.NewBufferString("somepayload")),
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid header for secret, but matching event",
+			Bitbucket: &triggersv1.BitBucketInterceptor{
+				SecretRef: &triggersv1.SecretRef{
+					SecretName: "mysecret",
+					SecretKey:  "token",
+				},
+				EventTypes: []string{"pr:opened", "repo:refs_changed"},
+			},
+			args: args{
+				signature: "foo",
+				secret: &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "mysecret",
+					},
+					Data: map[string][]byte{
+						"token": []byte("secrettoken"),
+					},
+				},
+				eventType: "pr:opened",
+				payload:   ioutil.NopCloser(bytes.NewBufferString("somepayload")),
+			},
+			wantErr: true,
+		}, {
+			name:      "nil body does not panic",
+			Bitbucket: &triggersv1.BitBucketInterceptor{},
+			args: args{
+				payload:   nil,
+				signature: "foo",
+			},
+			want:    []byte{},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, _ := rtesting.SetupFakeContext(t)
+			logger, _ := logging.NewLogger("", "")
+			kubeClient := fakekubeclient.Get(ctx)
+			request := &http.Request{
+				Body: tt.args.payload,
+				Header: http.Header{
+					"Content-Type": []string{"application/json"},
+				},
+			}
+			if tt.args.eventType != "" {
+				request.Header.Add("X-Event-Key", tt.args.eventType)
+			}
+			if tt.args.signature != "" {
+				request.Header.Add("X-Hub-Signature", tt.args.signature)
+			}
+			if tt.args.secret != nil {
+				ns := tt.Bitbucket.SecretRef.Namespace
+				if ns == "" {
+					ns = metav1.NamespaceDefault
+				}
+				if _, err := kubeClient.CoreV1().Secrets(ns).Create(tt.args.secret); err != nil {
+					t.Error(err)
+				}
+			}
+			w := &Interceptor{
+				KubeClientSet: kubeClient,
+				Bitbucket:     tt.Bitbucket,
+				Logger:        logger,
+			}
+			resp, err := w.ExecuteTrigger(request)
+			if err != nil {
+				if !tt.wantErr {
+					t.Errorf("Interceptor.ExecuteTrigger() error = %v, wantErr %v", err, tt.wantErr)
+				}
+				return
+			}
+
+			got, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				t.Fatalf("error reading response body %v", err)
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
hmm...looks like we are using `reflect.DeepEqual` in this package and `cmp.Diff` in others. Would be nice to use `cmp.Diff` everywhere (can be a separate cleanup PR in the future!)

Reply:
`reflect.DeepEqual` was used in 3 more test cases so handled here only